fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintainer.java

```diff
@@ -240,6 +244,22 @@ private void insertField(DocumentEntry entry, final Document document) {
         }
     }
 
+    public <M extends Message> DocumentEntry extractGrouping(@Nullable FDBIndexableRecord<M> record){
```

Comment:
The restriction that the grouping key be a single string value seems unnecessarily strict.

The only code that does not look like it would just work by using whole tuples is `getWriterName` and that could even use hex-encoded bytes, since it is internal only and quite local.

Relatedly, it does not seem that `DocumentEntry` is particularly appropriate for the extracted grouping. I think that things could be simplified by only normalizing the grouped key and using the regular evaluator on the grouping key to determine its value, again as a tuple.

The restriction that there is only one grouping for the entire record may be okay for now, although it will disallow the case of a map-like nested repeated message with the group split in the middle.